src/main/java/com/appium/manager/CloudAppiumManager.java

```diff
@@ -0,0 +1,54 @@
+package com.appium.manager;
+
+import com.github.device.Device;
+
+import java.util.List;
+
+import static java.lang.System.getenv;
+import static java.text.MessageFormat.format;
+
+public class CloudAppiumManager implements IAppiumManager {
+    @Override
+    public void destroyAppiumNode(String host) {
+
+    }
+
+    @Override
+    public String getRemoteWDHubIP(String host) {
+        String url = "https://{0}:{1}@hub-cloud.browserstack.com/wd/hub";
+        if (host.contains("sauce")) {
+            url = "https://{0}:{1}@ondemand.eu-central-1.saucelabs.com/wd/hub";
```

Comment:
@saikrishna321 I believe this URL changes depending on their data center.

Reply:
Currently we did changes only for SauceLabs and BrowserStack.